### DISCUSSION TITLE
Remove deprecated rules in v6?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6008] [RFCs]
## Overview

Forking from https://github.com/typescript-eslint/typescript-eslint/pull/5886#issuecomment-1311097524 by @bmish: we have 5 deprecated rules still shipped to users:

* [@typescript-eslint/no-duplicate-imports](https://typescript-eslint.io/rules/no-duplicate-imports)
* [@typescript-eslint/no-implicit-any-catch](https://typescript-eslint.io/rules/no-implicit-any-catch)
* [@typescript-eslint/no-parameter-properties](https://typescript-eslint.io/rules/no-parameter-properties)
* [@typescript-eslint/sort-type-union-intersection-members](https://typescript-eslint.io/rules/sort-type-union-intersection-members)
* [@typescript-eslint/camelcase](https://typescript-eslint.io/rules/camelcase)

For v6, how about we remove them altogether?

**Comment by bradzacher** [comment]
> [@typescript-eslint/camelcase](https://typescript-eslint.io/rules/camelcase)

We don't ship this rule, it's just listed in the docs so that we can redirect people to naming-convention.
This is done on purpose so that we don't have people asking for an extension rule for camecase

**Comment by JoshuaKGoldberg** [comment]
I'm going to go ahead and turn this _back_ into an issue. Furthermore, I'd say that removing deprecated rules should be part of the process for any new major version. Added to the `generate-breaking-changes.mts` script in #5251.
